appProject/appProject/ProductScreen/AppleStoreViewController.swift

```diff
@@ -0,0 +1,128 @@
+// 
+//  appleStoreViewController.swift
+//  appProject
+//
+//  Created by Валентин Коскин on 11.10.2022.
+//
+
+import UIKit
+import WebKit
+/// экран открытия apple.com с нужным товаром
+final class AppleStoreViewController: UIViewController {
+    
+    // MARK: Constants
+    
+    private enum Constants {
+        static let backButtonName = "chevron.left"
+        static let forwardButtonName = "chevron.right"
+        static let shareButtonName = "square.and.arrow.up"
+    }
+
+    // MARK: Visual Components
+    
+    private lazy var browserWebView = makeWebView(link: appleStoreProduct.link)
+    
+    private lazy var managingToolBar = makeToolBar()
+  
+    private lazy var backButton = UIBarButtonItem(image: UIImage(systemName: Constants.backButtonName),
+                                                  style: .plain, target: self, action: #selector(goBackAction))
+    
+    private lazy var forwardButton = UIBarButtonItem(image: UIImage(systemName: Constants.forwardButtonName),
+                                                     style: .plain, target: self, action: #selector(goForwardAction))
+    
+    private lazy var refreshButton = UIBarButtonItem(barButtonSystemItem: .refresh, target: self,
+                                                     action: #selector(refreshAction))
+    
+    private lazy var shareButton = UIBarButtonItem(image: UIImage(systemName: Constants.shareButtonName),
+                                                   style: .plain, target: self, action: #selector(shareAction))
+    
+    private lazy var spacer = UIBarButtonItem(systemItem: .flexibleSpace)
+    
+    private var sharePageActivityViewController: UIActivityViewController?
+    
+    private var oberver: NSKeyValueObservation?
+    
+    private let progressView = UIProgressView()
+    
+    // MARK: Public Properties
+    
+    var appleStoreProduct = Product()
+    
+    // MARK: Life Cycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        confgUI()
+    }
+    
+    // MARK: Private Methods
+    
+    private func confgUI() {
+        view.backgroundColor = .systemBackground
+        view.addSubview(browserWebView)
+        browserWebView.uiDelegate = self
+        view = browserWebView
+        browserWebView.uiDelegate = self
+        managingToolBar.items = [backButton, forwardButton, spacer, refreshButton, shareButton]
+        browserWebView.addSubview(managingToolBar)
+        progressView.frame = CGRect(x: 110, y: 800, width: 190, height: 30)
+        browserWebView.addSubview(progressView)
+        oberver = browserWebView.observe(\.estimatedProgress, options: [.new]) { [self] _, _ in
```

Comment:
```suggestion
        oberver = browserWebView.observe(\.estimatedProgress, options: [.new]) { [ weak self ] _, _ in
```
https://stackoverflow.com/questions/49347331/is-weak-self-needed-in-nskeyvalueobservation-observe-closure